### PR TITLE
add DeepMoD_PC

### PR DESCRIPTION
**Pull request by Faezehabibi**

### CODE REVIEW

File: exhibits/DeepMoD_PC/train_deepmod.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    print("\r >epoch={}  L= {:.4f}| Sparse Weight: Wdx = {} | Wdy = {}".format(i,
+                                                                     loss_pred/T,
+                                                                     (deepmod.thresholding() * scale).T[0],
+                                                                     (deepmod.thresholding() * scale).T[1]), end="")
+    if i%100 == 0:
+        print()
+
+    cov_cria = (coeff_track - coeff).mean()
+    coeff_track = coeff
+    if jnp.abs(cov_cria) <= 5e-8 or i==n_epochs-1:
```
Comment (rxng8) [inline exhibits/DeepMoD_PC/train_deepmod.py]: if we have run all the epochs, and `cov_cria` has not converged yet, we cannot call it converged. So I think we should remove the `i == n_epochs - 1` condition and further optimize the model

File: exhibits/DeepMoD_PC/deepmod.py [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+    def prediction_process(self, input, target):
+        self.model.batch_set(len(input))
+        self.E1.weights.set(self.W1.weights.value.T)
+        self.E2.weights.set(self.W2.weights.value.T)
+
+        self.model.reset()
+        self.model.clamps(input, target)
+
+        z_codes = self.model._process(jnp.array([[self.dt * i, self.dt] for i in range(self.T)]))
```
Comment (rxng8) [inline exhibits/DeepMoD_PC/deepmod.py]: `z_codes` was not used, is this needed?
I think it's fine to leave it there if it serves the `._process` purpose, leaving a comment will clarify for the reader

File: exhibits/DeepMoD_PC/train_deepmod.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+deepmod = DeepMoD(key=key, ts=ts[:, None], dict_dim=feat_dim, lib_creator=library_creator,
+                  solver_name="l1", l1_ratio=0.5, eta=eta,
+                  in_dim=in_dim, h1_dim=h1_dim, h2_dim=h2_dim, out_dim=out_dim,
+                  batch_size=batch_size, threshold=threshold, scale=scale,
+                  w_fill=w_fill, lr=lr, lmbda=lasso_lmbda)
+
+
+coeff_track = 0
+for i in range(n_epochs):
+    coeff, loss_pred = deepmod.process(ts_scaled, X)
```
Comment (rxng8) [inline exhibits/DeepMoD_PC/train_deepmod.py:113]: The loss in this case is smaller than 0, you then checked if the absolute value of the loss is smaller than some constant (`5e-8`), is this behavior desired?
I asked that because normally, the loss is greater than 0, and often converge toward 0.

File: exhibits/DeepMoD_PC/train_deepmod.py (hunk 1 above)
Comment (Faezehabibi) [reply exhibits/DeepMoD_PC/train_deepmod.py]: Removed extra the `i == n_epochs - 1` condition

File: exhibits/DeepMoD_PC/deepmod.py (hunk 2 above)
Comment (Faezehabibi) [reply exhibits/DeepMoD_PC/deepmod.py]: Removed the extra `Z_codes`

File: exhibits/DeepMoD_PC/train_deepmod.py (hunk 3 above)
Comment (Faezehabibi) [reply exhibits/DeepMoD_PC/train_deepmod.py:113]: Wait, I do not check the convergence based on the loss, I do based on the coefficients. Would you check this again and let me know if your idea?

Comment (rxng8) [reply exhibits/DeepMoD_PC/train_deepmod.py:113]: Cool! I think this looks good now! I did try to run the code again, I think you put the stopping criteria too slow (`5e-8`) that it runs forever. I recommend `5e-6` or even `1e-5`
